### PR TITLE
add initial ttl language

### LINKED ISSUE #96: Consider including next update time claim for Status List
To provide guidance for caching it might make sense to include a next update time in the status list. Or does `exp` provide this guidance?

### PR DESCRIPTION
**Pull request by tplooker**
Closes #96 

Alternative to #49 as it had gone stale.

### CODE REVIEW

**Review by paulbastian — APPROVED** [review]

File: draft-ietf-oauth-status-list.md [hunk 1]
```diff
@@ -567,6 +572,21 @@ To indicate that the content is an CWT-based Status List:
...
+The "Claim Description", "Change Controller" and "Specification Documents" are common and equivalent for the JWT and CWT registries.
+The "Claim Key" and "Claim Value Types(s)" are for the CWT registry only.  The "Claim Name" is as defined for the CWT registry, not the
+JWT registry.  The "JWT Claim Name" is equivalent to the "Claim Name" in the JWT registry.
+
+ - Claim Name: ttl
+ - Claim Description: Time to Live
+ - JWT Claim Name: "ttl"
+ - Change Controller: IESG
+ - Specification Document(s): *this document*
+
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:589]: This section should be moved up to line ~450where "status" is defined for JWT

**Review by c2bo — CHANGES_REQUESTED** [review]

File: draft-ietf-oauth-status-list.md [hunk 2]
```diff
@@ -201,9 +201,10 @@ The following content applies to the JWT Header:
...
-* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for that Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
+* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for the Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
 * `iat`: REQUIRED. The `iat` (issued at) claim MUST specify the time at which the Status List Token was issued.
-* `exp`: OPTIONAL. The `exp` (expiration time) claim MAY convey the time at which it is considered expired by its issuer.
+* `exp`: OPTIONAL. The `exp` (expiration time) claim, if present, MUST specify the time at which the Status List Token is considered expired by its issuer.
+* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token may be cached by a consumer before a fresh copy is retrieved. The value of the claim MUST be a positive number.
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:207]:
```suggestion
* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token may be cached by a consumer before a fresh copy is retrieved. The value of the claim MUST be a positive number.
```

File: draft-ietf-oauth-status-list.md [hunk 3]
```diff
@@ -382,7 +383,11 @@ Resulting in the byte array and compressed/base64url encoded status list:
 TODO elaborate on risks of incorrect parsing/decoding leading to erroneous status data
 
 ## Cached and Stale status lists
-TODO consumers/Relying Party of the status list should be aware if they fetch the up-to-date data
+
+When consumers or verifiers of the Status List fetch the data, they need to be aware of its up-to-date status. The 'ttl' (time-to-live) claim
+in the Status List Token provides one mechanism for setting a maximum cache time for the fetched data. This property permits distribution of
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:388]:
```suggestion
in the Status List Token provides one mechanism for setting a maximum cache time for the fetched data. This property permits distribution of
```

File: draft-ietf-oauth-status-list.md [hunk 4]
```diff
@@ -382,7 +383,11 @@ Resulting in the byte array and compressed/base64url encoded status list:
 TODO elaborate on risks of incorrect parsing/decoding leading to erroneous status data
 
 ## Cached and Stale status lists
-TODO consumers/Relying Party of the status list should be aware if they fetch the up-to-date data
+
+When consumers or verifiers of the Status List fetch the data, they need to be aware of its up-to-date status. The 'ttl' (time-to-live) claim
+in the Status List Token provides one mechanism for setting a maximum cache time for the fetched data. This property permits distribution of 
+a status list to a CDN or other distribution mechanism while giving guidance to consumers of the status list on how often they need to fetch
```
Comment (c2bo) [inline draft-ietf-oauth-status-list.md:389]:
```suggestion
a status list to a CDN or other distribution mechanism while giving guidance to consumers of the status list on how often they need to fetch
```

File: draft-ietf-oauth-status-list.md [hunk 5]
```diff
@@ -201,9 +201,10 @@ The following content applies to the JWT Header:
...
-* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for that Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
+* `sub`: REQUIRED. The `sub` (subject) claim MUST specify a unique string identifier for the Status List Token. The value MUST be equal to that of the `uri` claim contained in the `status_list` claim of the Referenced Token.
 * `iat`: REQUIRED. The `iat` (issued at) claim MUST specify the time at which the Status List Token was issued.
-* `exp`: OPTIONAL. The `exp` (expiration time) claim MAY convey the time at which it is considered expired by its issuer.
+* `exp`: OPTIONAL. The `exp` (expiration time) claim, if present, MUST specify the time at which the Status List Token is considered expired by its issuer.
+* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token may be cached by a consumer before a fresh copy is retrieved. The value of the claim MUST be a positive number.
```
Comment (tplooker) [inline draft-ietf-oauth-status-list.md:207]:
```suggestion
* `ttl`: OPTIONAL. The `ttl` (time to live) claim, if present, MUST specify the maximum amount of time, in seconds, that the Status List Token can be cached by a consumer before a fresh copy SHOULD be retrieved. The value of the claim MUST be a positive number.
```

**Comment by c2bo** [comment]
I moved the claim registration up - we have to do CWT registration after the CBOR/CWT PR has been merged - i'll create an issue for this

**Review by c2bo — APPROVED** [review]

**Pull request merged**